Sources/Datadog/RUM/Instrumentation/Actions/UIKit/UIKitRUMUserActionsHandler.swift

```diff
@@ -48,33 +42,63 @@ internal class UIKitRUMUserActionsHandler: UIEventHandler {
             return
         }
 
-        if let rumAction = predicate.rumAction(targetView: actionTargetView) {
-            subscriber?.process(
-                command: RUMAddUserActionCommand(
-                    time: dateProvider.currentDate(),
-                    attributes: rumAction.attributes,
-                    actionType: .tap,
-                    name: rumAction.name
-                )
-            )
-        }
+        subscriber.process(command: command)
     }
 
-    // MARK: - Events Filtering
+    // MARK: - Event Processing
 
-    /// Returns the `UITouch` for given `event` only if the event describes the "single tap ended" interaction.
-    private func captureSingleTouch(event: UIEvent) -> UITouch? {
+    #if !os(tvOS)
+    private func command(from event: UIEvent) -> RUMAddUserActionCommand? {
```

Comment:
It won't be possible to run unit tests for both implementation on single Simulator, no? How about using OCP and passing this logic as configuration 🤔💭?
```swift
let tvOSCommandFactory: (UIEvent) -> RUMAddUserActionCommand = ...
let ioOSCommandFactory: (UIEvent) -> RUMAddUserActionCommand = ...

let handler = UIKitRUMUserActionsHandler(..., commandFactory:)
```
This way we'll only need to `#if`-compile the initialization of this handler, but both platforms logic can be tested with single Simulator.

Reply:
We can now test both logic on a single simulator. This also imply that both predicates, for `UITouch` and `UIPress` events are accessible on all platforms.